### PR TITLE
GC: Beta Client Package

### LINKED ISSUE #2086: CI:  Failure to Checkout with Beta Package Added
Failure during [CI](https://github.com/alcionai/corso/actions/runs/3883931965/jobs/6626321063)
This could be related to memory in the CI or another issue. 

### Blocks

- #2069 
- #2071 
- #2084

### LINKED ISSUE #2174: GC: Add `Beta` Servicer feature
- To complete Backup / Restore pipelines, a beta connector is required for Pages calls. 

##### Related to:
- #2169 
- #2071 
- #2173

### LINKED ISSUE #2223: GC: SharePoint: Minimal-SitePage Data Type
Create a minimal version of `SharePoint.Pages` into the repository until data type is introduced into v1.0 of `msgraph`. 
Data type needs to support the [Parsable](https://pkg.go.dev/github.com/microsoft/kiota-abstractions-go@v0.16.0/serialization#Parsable) interface.

### PR DESCRIPTION
**Pull request by dadams39**
## Description
To add needed features, Kiota-Generated Beta Connector created

## Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included

## Type of change

- [x] :sunflower: Feature

## Issue(s)

* closes  #2086<issue>
* closes #2174 
* closes #2223

## Test Plan
Compilation and current CI tests will fail if the package does not align with current operational standards. 
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/connector/sharepoint/site_page.go [hunk 1]
```diff
@@ -1,187 +0,0 @@
...
-import (
-	kioser "github.com/microsoft/kiota-abstractions-go/serialization"
-	"github.com/microsoftgraph/msgraph-sdk-go/models"
-)
-
-// SitePage provides operations to manage the minimal creation of a Site Page.
-// Altered from original: github.com/microsoftgraph/msgraph-beta-sdk-go/models
-// TODO:  remove when Issue #2086 resolved
-type SitePage struct {
-	models.BaseItem
```
Comment (dadams39) [inline src/internal/connector/sharepoint/site_page.go:12]: This file is already present in the Beta Package. Removed.

File: src/internal/connector/sharepoint/site_pageable.go [hunk 2]
```diff
@@ -1,24 +0,0 @@
...
-
-import (
-	"github.com/microsoft/kiota-abstractions-go/serialization"
-	"github.com/microsoftgraph/msgraph-sdk-go/models"
-)
-
-// SitePageable adjusted from msgraph-beta-sdk-go for temporary testing
-type SitePageable interface {
-	models.BaseItemable
-	serialization.Parsable
```
Comment (dadams39) [inline src/internal/connector/sharepoint/site_pageable.go:11]: This file is also present in Beta package. Removed

File: src/internal/connector/sharepoint/pageInfo.go [hunk 3]
```diff
@@ -3,13 +3,15 @@ package sharepoint
...
+	"github.com/microsoftgraph/msgraph-beta-sdk-go/models"
+
 	"github.com/alcionai/corso/src/pkg/backup/details"
 )
 
 // sharePointPageInfo propagates metadata from  the SharePoint Page data type
 // into searchable content.
 // Page Details: https://learn.microsoft.com/en-us/graph/api/resources/sitepage?view=graph-rest-beta
-func sharePointPageInfo(page SitePageable, size int64) *details.SharePointInfo {
+func sharePointPageInfo(page models.SitePageable, size int64) *details.SharePointInfo {
```
Comment (dadams39) [inline src/internal/connector/sharepoint/pageInfo.go:14]: Uses the data type already present in msgraph-beta-sdk-go

**Review by ryanfkeepers — APPROVED** [review]

**Comment by dadams39** [comment]
`Beta` is not production value: Build-times / serialization / and impact on downstream products is too great. 
- This will be the anchor for feature development.

**Comment by dadams39** [comment]
Process:
- Generated package 110k of line changes
- Removal of obvious site request which are found in v1.0 --> 800k lines of change 
  - It is important to consider that changes for the Site upload may require a multi-step restore workflow 
- Removed models that are not necessary for `SharePoint.Page GET / POST ` commands --> 15k of changes.
- Will reduce tomorrow to: `Get, Select, Post, Publish`

File: src/internal/connector/graph/betasdk/beta_client.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+	})
+
+	if m.requestAdapter.GetBaseUrl() == "" {
+		m.requestAdapter.SetBaseUrl("https://graph.microsoft.com/beta")
+	}
+	return m
+}
+
+// SitesById provides operations to manage the collection of site entities.
+func (m *BetaClient) SitesById(id string) *i1a3c1a5501c5e41b7fd169f2d4c768dce9b096ac28fb5431bf02afcc57295411.SiteItemRequestBuilder {
```
Comment (ryanfkeepers) [inline src/internal/connector/graph/betasdk/beta_client.go:77]: Can this get moved to `connector/discovery`?  That's where we're currently housing the api layer for resource owners.

Comment (ryanfkeepers) [inline src/internal/connector/graph/betasdk/beta_client.go:77]: What is the ID in this case?  Also, the name `SitesById` makes it sound like the one ID returns many sites.  Is that true?

File: CHANGELOG.md [hunk 5]
```diff
@@ -12,6 +12,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 
 - Check if the user specified for an exchange backup operation has a mailbox.
+
+### Changed
+
+- msgraph-beta-sdk-go replaces msgraph-sdk-go for new features. This can lead to long build times.
```
Comment (ryanfkeepers) [inline CHANGELOG.md:18]: Is this still true if we're using a minimized/sliced version of the beta package?

**Review by ryanfkeepers — COMMENTED** [review]
Some minor nits about the SitesById func.  Otherwise, I can't possibly review all of the new files, but the other modifications look good.

File: src/internal/connector/graph/betasdk/beta_client.go (hunk 4 above)
Comment (dadams39) [reply src/internal/connector/graph/betasdk/beta_client.go:77]: No. All of the objects within the current `betasdk` layer are auto-generated. Within #2298, it is possible to add custom information there specifically to the `beta_service.go` file. As this file will be propagated for testing in the future.

Comment (dadams39) [reply src/internal/connector/graph/betasdk/beta_client.go:77]: The Kiota naming conventions favor plural notation when generated. We have implemented this code in other areas:
```go
	gc.Service.Client().UsersById()
	gc.Service.Client().UsersById().MessagesById()
```

File: CHANGELOG.md (hunk 5 above)
Comment (dadams39) [reply CHANGELOG.md:18]: This is not true. I have recompiled the code on several occasions. As of now, we are not experiencing the build time errors that were presented previously. The main reason is all the abstract kiota libraries are imported with msgraph. The build takes approx. 90 minutes with a cache miss. This is similar to our current scenario.

Comment (dadams39) [reply CHANGELOG.md]: Resovled within [cd51aae](https://github.com/alcionai/corso/pull/2205/commits/cd51aaea7c693a930cf996604f7b537926b6d2d0)

**Pull request merged**